instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import './CommentSection.css'
+import moment from 'moment'
+
+export default class CommentSection extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      likes: this.props.likes,
+      isLiked: false,
+      comments: [],
+      user: localStorage.getItem("user"),
+      newComment: "",
+      heart: "far fa-heart fa-2x",
+      isLiked: false
+    }
+  }
+
+  componentDidMount() {
+    this.setState({comments: this.props.dummyData})
+  }
+
+  isLikedHandler = () => {
+    if (!this.state.isLiked) { 
+      let likes = this.state.likes + 1; 
+      this.setState({
+        likes,
+        isLiked: true,
+        heart: "fas fa-heart fa-2x"
+      })
+    }
+
+    if (this.state.isLiked) {
+      let likes = this.state.likes - 1;
+      this.setState({
+        likes,
+        isLiked: false,
+        heart: "far fa-heart fa-2x"
+      })
+    }
+  }
+
+  addNewComment = e => {
+    e.preventDefault()
+    let userInput = {
+      username: this.state.user,
+      text: this.state.newComment
+    }
+
+    const comments = this.state.comments.slice()
```

Comment:
Although it works, this is an outdated pattern. The same can be achieved using the spread operator.
```suggestion
   this.setState({
      comments: [...this.state.comments, userInput],
      newComment: ""
    });
```